src/extensionsIntegrated/CustomSnippets/helper.js

```diff
@@ -0,0 +1,778 @@
+        }
+    }
+
+    function validateAbbrInput(e, abbrBox) {
+        // Allow keyboard shortcuts and navigation keys
+        if (e.ctrlKey || e.metaKey || e.altKey) {
+            return;
+        }
+
+        // Allow navigation and function keys
+        if (ALLOWED_NAVIGATION_KEYS.includes(e.key)) {
+            return;
+        }
+
+        // Prevent space character
+        if (e.key === " ") {
+            e.preventDefault();
+
+            // Determine if this is the edit form or new form
+            const isEditForm = abbrBox.id === "edit-abbr-box";
+            const inputId = isEditForm ? "edit-abbr-box" : "abbr-box";
+            const wrapperId = isEditForm ? "edit-abbr-box-wrapper" : "abbr-box-wrapper";
+            const errorId = isEditForm ? "edit-abbreviation-space-error" : "abbreviation-space-error";
+
+            UIHelper.showError(inputId, wrapperId, "Space is not accepted as a valid abbreviation character.", errorId);
+            return;
+        }
+
+        // Check for character limit (30 characters) - only for printable characters
+        if (
+            abbrBox.value.length >= 30 &&
+            e.key.length === 1 &&
+            e.key.match(/[a-zA-Z0-9!@#$%^&*()_+\-=\[\]{};':"\\|,.<>\/?]/)
+        ) {
+            e.preventDefault();
+
+            // Determine if this is the edit form or new form
+            const isEditForm = abbrBox.id === "edit-abbr-box";
+            const inputId = isEditForm ? "edit-abbr-box" : "abbr-box";
+            const wrapperId = isEditForm ? "edit-abbr-box-wrapper" : "abbr-box-wrapper";
+            const errorId = isEditForm ? "edit-abbreviation-length-error" : "abbreviation-length-error";
+
+            UIHelper.showError(inputId, wrapperId, "Abbreviation cannot be more than 30 characters.", errorId);
+        }
+    }
+
+    function validateDescInput(e, descBox) {
+        // Allow keyboard shortcuts and navigation keys
+        if (e.ctrlKey || e.metaKey || e.altKey) {
+            return;
+        }
+
+        // Allow navigation and function keys
+        if (ALLOWED_NAVIGATION_KEYS.includes(e.key)) {
+            return;
+        }
+
+        // Check for character limit (80 characters) - only for printable characters (spaces allowed)
+        if (
+            descBox.value.length >= 80 &&
+            e.key.length === 1 &&
+            e.key.match(/[a-zA-Z0-9!@#$%^&*()_+\-=\[\]{};':"\\|,.<>\/?\ ]/)
+        ) {
+            e.preventDefault();
+
+            // Determine if this is the edit form or new form
+            const isEditForm = descBox.id === "edit-desc-box";
+            const inputId = isEditForm ? "edit-desc-box" : "desc-box";
+            const wrapperId = isEditForm ? "edit-desc-box-wrapper" : "desc-box-wrapper";
+            const errorId = isEditForm ? "edit-description-length-error" : "description-length-error";
+
+            UIHelper.showError(inputId, wrapperId, "Description cannot be more than 80 characters.", errorId);
```

Comment:
move all UI strings to `root/strings.js`. The strings are all in English currently and not localized.